src/Widgets/SearchView.vala

```diff
@@ -28,9 +28,39 @@ namespace Slingshot.Widgets {
         public signal void start_search (Synapse.SearchMatch search_match, Synapse.Match? target);
         public signal void app_launched ();
 
+        private class CycleListBox : Gtk.ListBox {
+            public override void move_cursor (Gtk.MovementStep step, int count) {
+                unowned Gtk.ListBoxRow selected = get_selected_row ();
+
+                if (step != Gtk.MovementStep.DISPLAY_LINES || selected == null) {
+                    base.move_cursor (step, count);
+                    return;
+                }
+
+                uint n_children = get_children ().length ();
+
+                int current = selected.get_index ();
+                int target = 0;
+
+                if (count < 0 && current <= -count - 1) {
+                    target = (int)n_children - 1;
+                } else if (count > 0 && current >= n_children - count) {
+                    target = 0;
+                } else {
+                    target = current + count;
+                }
+
```

Comment:
Assuming count may exceed 1 (or less than -1), then this swallows the extra steps when cycling.
If count is always either 1 or -1 (or treated as such)  then the code can be simplified.

`target` can be initialized as ` current + count`.  The if clauses can then be written as:

`if (target < 0) { target = n_children - 1;}`
and
`if (target >= n_children) {target = 0;}`

which is a little clearer.